frontend/src/components/question-creation/question-elements/ImageElement.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useContext, useEffect, useRef, useState } from "react";
+import { useMutation } from "@apollo/client";
+import { Button, Flex, Image, Input } from "@chakra-ui/react";
+
+import { UPLOAD_TEST_IMAGE } from "../../../APIClients/mutations/TestMutations";
+import { UploadOutlineIcon } from "../../../assets/icons";
+import { JUMP_MATH_LOGO } from "../../../assets/images";
+import QuestionEditorContext from "../../../contexts/QuestionEditorContext";
+import { ImageMetadata } from "../../../types/QuestionMetadataTypes";
+import { updatedQuestionElement } from "../../../utils/QuestionUtils";
+
+interface ImageElementProps {
+  id: string;
+  data: ImageMetadata;
+}
+
+const ImageElement = ({ id, data }: ImageElementProps): React.ReactElement => {
+  const [imageMetadata, setImageMetadata] = useState<ImageMetadata>(data);
+  const [error, setError] = useState<string>("");
+  const [uploadTestImage] = useMutation<{ uploadTestImage: ImageMetadata }>(
+    UPLOAD_TEST_IMAGE,
+    {
+      onCompleted: (res) => {
+        setImageMetadata(res.uploadTestImage);
+      },
+      onError: (err) => {
+        setError("Error uploading image.");
+      },
+    },
+  );
+  const inputFile = useRef<HTMLInputElement>(null);
+
+  const { setQuestionElements } = useContext(QuestionEditorContext);
+
+  const updateImageElement = async (e: { target: HTMLInputElement }) => {
+    setError("");
+    const fileList: FileList | null = e.target.files;
+    if (!fileList || !fileList[0]) return;
+
+    const fileSize = fileList[0].size / 1024 / 1024;
```

Comment:
nit: is this conversion to MB? maybe we can refactor to a function in `GeneralUtils` for readability?